### PR TITLE
[ALLUXIO-2460] Remote hdfs hint

### PR DESCRIPTION
**Pull request by calvinjia**
https://alluxio.atlassian.net/browse/ALLUXIO-2460

### CODE REVIEW

File: underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java [hunk 1]
```diff
@@ -197,6 +197,11 @@ public Object getConf() {
   @Override
   public List<String> getFileLocations(String path, FileLocationOptions options)
       throws IOException {
+    // If the user has hinted the underlying storage nodes are not co-located with Alluxio
+    // workers, short circuit without querying the locations
+    if (Configuration.getBoolean(PropertyKey.UNDERFS_HDFS_REMOTE)) {
+      return null;
```
Comment (aaudiber) [inline underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java]: what's the difference between returning null and returning an empty list?

Comment (calvinjia) [inline underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java]: There isn't a difference at the moment.

Comment (aaudiber) [inline underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java]: I'd prefer to return an empty list then for consistency.

Comment (aaudiber) [inline underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java]: We should probably specify the extraordinary return values in the interface javadoc

**Pull request merged**